packages/flutter_tools/lib/src/desktop_device.dart

```diff
@@ -202,6 +204,92 @@ abstract class DesktopDevice extends Device {
   /// Called after a process is attached, allowing any device-specific extra
   /// steps to be run.
   void onAttached(ApplicationPackage package, BuildMode buildMode, Process process) {}
+
+  /// Computes a set of environment variables used to pass debugging information
+  /// to the engine without interfering with application level command line
+  /// arguments.
+  ///
+  /// The format of the environment variables is:
+  ///   * FLUTTER_ENGINE_SWITCHES to the number of switches.
+  ///   * FLUTTER_ENGINE_SWITCH_<N> (indexing from 1) to the individual switches.
+  Map<String, String> _computeEnvironment(DebuggingOptions debuggingOptions, bool traceStartup, String route) {
+    int flags = 1;
+    final Map<String, String> environment = <String, String>{};
+
+    void addFlag(String value) {
+      environment['FLUTTER_ENGINE_SWITCH_$flags'] = value;
+      flags += 1;
+    }
+    void finish() {
+      environment['FLUTTER_ENGINE_SWITCHES'] = (flags - 1).toString();
+    }
+
+    addFlag('enable-dart-profiling=true');
+    addFlag('enable-background-compilation=true');
+
+    if (traceStartup) {
+      addFlag('trace-startup=true');
+    }
+    if (route != null) {
+      addFlag('route=$route');
+    }
+    if (debuggingOptions.enableSoftwareRendering) {
+      addFlag('enable-software-rendering=true');
```

Comment:
This won't actually work currently, but I guess it's harmless to pass anyway, and then it'll be in place already when implemented on the engine side.

Reply:
Do we have an issue filled? I could add that in a comment

Reply:
Actually, now that I look closer it might be that this is internal enough that it does work. I was thinking of the embedding using software mode at the embedder.h level, which may not be the same thing. Let's just leave this as-is.